### PR TITLE
Review request for Microsoft.ConfidentialLedger to add version preview/2024-01-26

### LINKED ISSUE #28064: [Azure Confidential Computing Ledger - Azure Confidential Ledger] API Review
New API Review meeting has been requested.

**Service Name**: Azure Confidential Computing Ledger - Azure Confidential Ledger
**Review Created By**: Tai Chou
**Review Date**: 03/12/2024 01:00 PM PT
**Release Plan**: [](https://web.powerapps.com/apps/821ab569-ae60-420d-8264-d7b5d5ef734c?release-plan-id=)
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/27564
**Hero Scenarios Link**: [here](https://microsoft-my.sharepoint-df.com/:w:/g/personal/tachou_microsoft_com1/EVUirfTSrlhJslGy0i_eCOgBP_wb0gDTlBZOBynm5wWoCw?e=XHbCX5)
**Core Concepts Doc Link**: [here](https://learn.microsoft.com/en-us/azure/confidential-ledger/architecture)

**Description**: Adding a small GET api to list all current ledger users

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=047289d3-5fda-ee11-904c-000d3a357edf)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by taicchoumsft**
# Data Plane API - Pull Request
Confidential Ledger is adding a new API to list users. A new preview API (2024-01-26-preview) is added, and a new API to list users 

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue: https://msazure.visualstudio.com/One/_workitems/edit/26168810

Is this review for (select one):

- [ ] a private preview
- [X] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document: [Doc](https://microsoft-my.sharepoint-df.com/:w:/g/personal/tachou_microsoft_com1/EVUirfTSrlhJslGy0i_eCOgBP_wb0gDTlBZOBynm5wWoCw?e=oaZcv4)
* Previous API Spec Doc:
* Updated paths: /app/users

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

fix https://github.com/Azure/azure-rest-api-specs/issues/28064

### CODE REVIEW

**Comment by taicchoumsft** [comment]
The required check Breaking Change(Cross-Version) is failing due to changes that are not related to this new API version (see [details](https://github.com/Azure/azure-rest-api-specs/pull/27564/checks?check_run_id=23128861559)).

Given that the https://github.com/Azure/azure-rest-api-specs/pull/21659 also had [the same detected breaking changes](https://github.com/Azure/azure-rest-api-specs/pull/21659/checks?check_run_id=13005052549) and was eventually approved, I've added the BreakingChange-Approved-Previously label.

**Comment by taicchoumsft** [comment]
@DominikMe , @johanste , the changes have been approved by the API review board, and the current build check breaks are not caused by this change, see [this comment](https://github.com/Azure/azure-rest-api-specs/pull/27564#issuecomment-2021643115).  Awaiting your reviews, thank you!

**Comment by taicchoumsft** [comment]
@DominikMe , @johanste , can we request for a review on this PR?  Is there anything we need to do to move this along? Thanks in advance!

File: specification/confidentialledger/data-plane/Microsoft.ConfidentialLedger/preview/2024-01-26-preview/confidentialledger.json [hunk 1]
```diff
@@ -0,0 +1,1122 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      }
+    },
+    "/app/users/": {
```
Comment (johanste) [inline specification/confidentialledger/data-plane/Microsoft.ConfidentialLedger/preview/2024-01-26-preview/confidentialledger.json:392]:
```suggestion
    "/app/users": {
```

**Review by johanste — APPROVED** [review]
Please update/remove the need for the trailing slash on the new list operation.

**Review by johanste — APPROVED** [review]

**Pull request merged**